From: Rick Jones <rick.jones2@hp.com>
Date: Mon, 7 Nov 2011 13:29:27 +0000
Subject: [PATCH 01/58] sweep the floors and convert some .get_drvinfo
 routines to strlcpy

commit 68aad78c5023b8aa82da99b47f9d8cf40e8ca453 upstream.

Per the mention made by Ben Hutchings that strlcpy is now the preferred
string copy routine for a .get_drvinfo routine, do a bit of floor
sweeping and convert some of the as-yet unconverted ethernet drivers to
it.

Signed-off-by: Rick Jones <rick.jones2@hp.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[bwh: Restrict to drivers/net/ethernet/emulex/benet/]
---
 drivers/net/ethernet/emulex/benet/be_ethtool.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

Index: linux-3.2.46/drivers/net/ethernet/emulex/benet/be_ethtool.c
===================================================================
--- linux-3.2.46.orig/drivers/net/ethernet/emulex/benet/be_ethtool.c	2013-07-26 18:55:27.000000000 +0000
+++ linux-3.2.46/drivers/net/ethernet/emulex/benet/be_ethtool.c	2013-07-26 19:20:51.000000000 +0000
@@ -127,8 +127,8 @@
 	memset(fw_on_flash, 0 , sizeof(fw_on_flash));
 	be_cmd_get_fw_ver(adapter, adapter->fw_ver, fw_on_flash);
 
-	strcpy(drvinfo->driver, DRV_NAME);
-	strcpy(drvinfo->version, DRV_VER);
+	strlcpy(drvinfo->driver, DRV_NAME, sizeof(drvinfo->driver));
+	strlcpy(drvinfo->version, DRV_VER, sizeof(drvinfo->version));
 	strncpy(drvinfo->fw_version, adapter->fw_ver, FW_VER_LEN);
 	if (memcmp(adapter->fw_ver, fw_on_flash, FW_VER_LEN) != 0) {
 		strcat(drvinfo->fw_version, " [");
@@ -136,7 +136,8 @@
 		strcat(drvinfo->fw_version, "]");
 	}
 
-	strcpy(drvinfo->bus_info, pci_name(adapter->pdev));
+	strlcpy(drvinfo->bus_info, pci_name(adapter->pdev),
+		sizeof(drvinfo->bus_info));
 	drvinfo->testinfo_len = 0;
 	drvinfo->regdump_len = 0;
 	drvinfo->eedump_len = 0;
